File: content/realm/examples/java/sync/app/src/main/java/com/mongodb/realm/examples/model/Turtle.kt
package com.mongodb.realm.examples.model
// :snippet-start: complete
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId

open class Turtle(var name : String? = null, var age : Int = 0): RealmObject() {
    @PrimaryKey
    var _id : ObjectId = ObjectId()
    var owner : TurtleEnthusiast? = null
}
// :snippet-end:

File: content/realm/examples/react-native/legacy/__tests__/ts/Models/Turtle.ts
import Realm, {BSON, ObjectSchema} from 'realm';
import Person from './Person';

class Turtle extends Realm.Object<Turtle> {
  _id!: BSON.ObjectId;
  name!: string;
  age!: number;
  owner?: Person;

  static schema: ObjectSchema = {
    name: 'Turtle',
    properties: {
      _id: 'objectId',
      name: 'string',
      owner: 'Person?',
      age: 'int',
    },
    primaryKey: '_id',
  };
}

export default Turtle;
